docker/runtime-entrypoint.sh

```diff
@@ -1,82 +1,95 @@
 #!/usr/bin/env bash
 
-# show the commands we are executing
+# Show the commands we are executing
 set -x
-# exit if a command fails
+# Exit if a command fails
 set -e
-# exit if any command in a pipeline fails
+# Exit if any command in a pipeline fails
 set -o pipefail
 
-echo "Config variables:"
-echo "NETWORK=$NETWORK"
-echo "RPC_PORT=$RPC_PORT"
-echo "LOG_FILE=$LOG_FILE"
-
-echo "Config location:"
-echo "ZEBRA_CONF_DIR=$ZEBRA_CONF_DIR"
-echo "ZEBRA_CONF_FILE=$ZEBRA_CONF_FILE"
-echo "ZEBRA_CONF_PATH=$ZEBRA_CONF_PATH"
+# Set this to change the default cached state directory
+# Path and name of the config file
+: "${ZEBRA_CONF_DIR:='/etc/zebrad'}"
+: "${ZEBRA_CONF_FILE:='zebrad.toml'}"
+if [[ -n "$ZEBRA_CONF_DIR" ]] && [[ -n "$ZEBRA_CONF_FILE" ]]; then
+    ZEBRA_CONF_PATH="$ZEBRA_CONF_DIR/$ZEBRA_CONF_FILE"
+fi
 
-echo "Other variables:"
-echo "SHORT_SHA=$SHORT_SHA"
-echo "SENTRY_DSN=$SENTRY_DSN"
+# [network]
+: "${NETWORK:='Mainnet'}"
+: "${ZEBRA_LISTEN_ADDR:='127.0.0.1'}"
+# [consensus]
+: "${ZEBRA_CHECKPOINT_SYNC:='true'}"
+# [state]
+: "${ZEBRA_CACHED_STATE_DIR:='/var/cache/zebrad-cache'}"
+: "${LOG_COLOR:='false'}"
+# [rpc]
+: "${RPC_LISTEN_ADDR:='127.0.0.1'}"
 
 # Create the conf path and file if it does not exist.
-mkdir -p "$ZEBRA_CONF_DIR"
-touch "$ZEBRA_CONF_PATH"
+if [[ -n "$ZEBRA_CONF_PATH" ]]; then
+    mkdir -p "$ZEBRA_CONF_DIR"
+    touch "$ZEBRA_CONF_FILE"
+fi
 
 # Populate `zebrad.toml` before starting zebrad, using the environmental
-# variables set by the Dockerfile.
+# variables set by the Dockerfile or the user.
 #
 # We disable most ports by default, so the default config is secure.
 # Users have to opt-in to additional functionality by setting environmental variables.
-#
-# TODO:
-#  - make `cache_dir`, `metrics.endpoint_addr`, and `tracing.endpoint_addr` into Docker arguments
-#  - add an $EXTRA_CONFIG or $REPLACEMENT_CONFIG environmental variable
+if [[ ! -f "$ZEBRA_CONF_PATH" ]]; then
 cat <<EOF > "$ZEBRA_CONF_PATH"
 [network]
 network = "$NETWORK"
-listen_addr = "0.0.0.0"
+listen_addr = $ZEBRA_LISTEN_ADDR:8233
```

Comment:
8233 is the default port for mainnet. If it is not supplied, Zebra automatically chooses the default port for the configured network.

Changing this port would be a breaking change for testnet.

https://doc.zebra.zfnd.org/zebra_network/struct.Config.html#structfield.listen_addr

```suggestion
listen_addr = $ZEBRA_LISTEN_ADDR
```